README.md

Contao Extension "newsgallery"
==============================


## Intro ##

This is an extension for the [Contao Open Source CMS][1]. If you are not a software developer, please download or install from the [Contao Extension Repository][2].


## Description ##

Using this extension, you can add an image gallery (identical to the content element) to each news item.

**In Contao 3, this extension is no longer working! Every news item can have unlimited content elements by default, including galleries.**


## Usage ##

After installation, you will see an additional section in every news item, where you can add your images. You must also choose the new news template *news\_full\_gallery* in your news reader frontend module. If you want to show the gallery in other news templates, simply apply the differences.


## Dependencies ##

The Contao Core extension "news" must obviously be enabled. There are no other external dependencies.

[1]: http://contao.org/
[2]: http://contao.org/en/extension-list.html


CHANGELOG.md

Contao Extension "newsgallery"
==============================


Version 1.0.6 stable (2012-11-13)
---------------------------------
- Updated for Contao 2.11
- Additional image-size options added
- Added changelog and readme


Version 1.0.5 stable (2012-01-23)
---------------------------------
- Added templates for Contao 2.10
- Removed template for 2.7


Version 1.0.4 stable (2010-03-10)
---------------------------------
Release 1.0.4 now supports random sorting and image cutting options in TYPOlight 2.8

**Attention:** For TYPOlight 2.7 you have to choose the "news_full_gallery_2.7" template!	


Version 1.0.3 stable (2009-10-15)
---------------------------------
Release 1.0.3 shows an icon to each news item in the backend if there is a gallery attached.	


Version 1.0.2 stable (2009-05-26)
---------------------------------
Release 1.0.2 has been optimized for TYPOlight 2.7 but still works with 2.6	


Version 1.0.2 stable (2009-05-26)
---------------------------------
Release 1.0.2 has been optimized for TYPOlight 2.7 but still works with 2.6	


Version 1.0.1 stable (2009-02-01)
---------------------------------
Version 1.0.1 adds a headline to the gallery which will be displayed in frontend like for content elements.	


Version 1.0.0 stable (2009-01-30)
---------------------------------
This feature has been requested so often I tought I should take care of it. It took me less than an hour to implement... and I don't need it ;-)	

NewsGallery.php
<?php if (!defined('TL_ROOT')) die('You cannot access this file directly!');

/**
 * Contao Open Source CMS
 * Copyright (C) 2005-2011 Leo Feyer
 *
 * Formerly known as TYPOlight Open Source CMS.
 *
 * This program is free software: you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation, either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this program. If not, please visit the Free
 * Software Foundation website at <http://www.gnu.org/licenses/>.
 *
 * PHP version 5
 * @copyright  Andreas Schempp 2009-2012
 * @author     Andreas Schempp <andreas@schempp.ch>
 * @license    http://opensource.org/licenses/lgpl-3.0.html
 */


class NewsGallery extends ContentGallery
{

	public function __construct($objTemplate)
	{
		$this->import('Config');
		$this->import('Input');
		$this->import('Environment');
		$this->import('Session');
		$this->import('Database');

		// Default values (not available in news gallery)
		$this->useHomeDir = false;
		$this->perPage = 0;
		$this->id = $objTemplate->id;
		$this->type = 'news_gallery';

		// Apply news gallery settings
		$arrHeadline = deserialize($objTemplate->gal_headline);
		$this->headline = is_array($arrHeadline) ? $arrHeadline['value'] : $arrHeadline;
		$this->hl = is_array($arrHeadline) ? $arrHeadline['unit'] : 'h1';
		$this->multiSRC = $objTemplate->multiSRC;
		$this->sortBy = $objTemplate->sortBy;
		$this->perRow = $objTemplate->perRow;
		$this->imagemargin = $objTemplate->gal_imagemargin;
		$this->size = $objTemplate->gal_size;
		$this->fullsize = $objTemplate->gal_fullsize;
		$this->numberOfItems = $objTemplate->gal_numberOfItems;
		$this->perPage = $objTemplate->gal_perPage;
	}
}



languages/pl/modules.php
<?php if (!defined('TL_ROOT')) die('You cannot access this file directly!');

/**
 * Contao Open Source CMS
 * Copyright (C) 2005-2011 Leo Feyer
 *
 * Formerly known as TYPOlight Open Source CMS.
 *
 * This program is free software: you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation, either
 * version 3 of the License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this program. If not, please visit the Free
 * Software Foundation website at <http://www.gnu.org/licenses/>.
 *
 * PHP version 5
 * @copyright  Andreas Schempp 2009-2012
 * @author     Bartosz Talarek <bartosz.talarek@gmail.com>
 * @license    http://opensource.org/licenses/lgpl-3.0.html
 */
 
 
/**
 * Extension folder
 */
$GLOBALS['TL_LANG']['MOD']['newsgallery'] = array('Galeria obrazków');



dca/tl_news.php
<?php if (!defined('TL_ROOT')) die('You cannot access this file directly!');

/**
 * Contao Open Source CMS
 * Copyright (C) 2005-2011 Leo Feyer
 *
 * Formerly known as TYPOlight Open Source CMS.
 *
 * This program is free software: you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation, either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this program. If not, please visit the Free
 * Software Foundation website at <http://www.gnu.org/licenses/>.
 *
 * PHP version 5
 * @copyright  Andreas Schempp 2009-2012
 * @author     Andreas Schempp <andreas@schempp.ch>
 * @license    http://opensource.org/licenses/lgpl-3.0.html
 */


/**
 * Listing
 */
$GLOBALS['TL_DCA']['tl_news']['list']['sorting']['child_record_callback'] = array('tl_news_gallery', 'listNewsArticles');


/**
 * Palettes
 */
$GLOBALS['TL_DCA']['tl_news']['palettes']['__selector__'][]		= 'addGallery';
$GLOBALS['TL_DCA']['tl_news']['subpalettes']['addGallery']		= 'gal_headline,multiSRC,gal_size,gal_imagemargin,perRow,gal_perPage,gal_numberOfItems,sortBy,gal_fullsize';
foreach($GLOBALS['TL_DCA']['tl_news']['palettes'] as $k => $v)
{
	$GLOBALS['TL_DCA']['tl_news']['palettes'][$k] = str_replace('addImage;', 'addImage;{gallery_legend:hide},addGallery;', $GLOBALS['TL_DCA']['tl_news']['palettes'][$k]);
}


/**
 * Fields
 */
$GLOBALS['TL_DCA']['tl_news']['fields']['addGallery'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['addGallery'],
	'exclude'                 => true,
	'filter'                  => true,
	'inputType'               => 'checkbox',
	'eval'                    => array('submitOnChange'=>true)
);

$GLOBALS['TL_DCA']['tl_news']['fields']['gal_headline'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['gal_headline'],
	'exclude'                 => true,
	'inputType'               => 'inputUnit',
	'options'                 => array('h1', 'h2', 'h3', 'h4', 'h5', 'h6'),
	'eval'                    => array('maxlength'=>255),
);

$GLOBALS['TL_DCA']['tl_news']['fields']['multiSRC'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['multiSRC'],
	'exclude'                 => true,
	'inputType'               => 'fileTree',
	'eval'                    => array('fieldType'=>'checkbox', 'files'=>true, 'mandatory'=>true),
);

$GLOBALS['TL_DCA']['tl_news']['fields']['perRow'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['perRow'],
	'default'                 => 4,
	'exclude'                 => true,
	'inputType'               => 'select',
	'options'                 => array(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12),
	'eval'                    => array('tl_class'=>'w50'),
);

$GLOBALS['TL_DCA']['tl_news']['fields']['sortBy'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['sortBy'],
	'exclude'                 => true,
	'inputType'               => 'select',
	'options'                 => array('name_asc', 'name_desc', 'date_asc', 'date_desc', 'meta'),
	'reference'               => &$GLOBALS['TL_LANG']['tl_news'],
	'eval'                    => array('tl_class'=>'w50'),
);

$GLOBALS['TL_DCA']['tl_news']['fields']['gal_imagemargin'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['gal_imagemargin'],
	'exclude'                 => true,
	'inputType'               => 'trbl',
	'options'                 => array('px', '%', 'em', 'pt', 'pc', 'in', 'cm', 'mm'),
	'eval'                    => array('includeBlankOption'=>true, 'tl_class'=>'w50'),
);

$GLOBALS['TL_DCA']['tl_news']['fields']['gal_size'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['gal_size'],
	'exclude'                 => true,
	'inputType'               => 'imageSize',
	'options'                 => $GLOBALS['TL_CROP'],
	'reference'               => &$GLOBALS['TL_LANG']['MSC'],
	'eval'                    => array('rgxp'=>'digit', 'nospace'=>true, 'tl_class'=>'w50'),
);

$GLOBALS['TL_DCA']['tl_news']['fields']['gal_fullsize'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_news']['gal_fullsize'],
	'exclude'                 => true,
	'inputType'               => 'checkbox',
	'eval'                    => array('tl_class'=>'w50'),
);

$GLOBALS['TL_DCA']['tl_news']['fields']['gal_perPage'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_content']['perPage'],
	'exclude'                 => true,
	'inputType'               => 'text',
	'eval'                    => array('rgxp'=>'digit', 'tl_class'=>'w50')
);

$GLOBALS['TL_DCA']['tl_news']['fields']['gal_numberOfItems'] = array
(
	'label'                   => &$GLOBALS['TL_LANG']['tl_content']['numberOfItems'],
	'exclude'                 => true,
	'inputType'               => 'text',
	'eval'                    => array('rgxp'=>'digit', 'tl_class'=>'w50')
);

if (version_compare(VERSION, '2.7', '>'))
{
	$GLOBALS['TL_DCA']['tl_news']['fields']['sortBy']['options'][] = 'random';
}



class tl_news_gallery extends Backend
{

	public function listNewsArticles($arrRow)
	{
		$key = $arrRow['published'] ? 'published' : 'unpublished';
		$date = $this->parseDate($GLOBALS['TL_CONFIG']['datimFormat'], $arrRow['date']);

		return '
<div class="cte_type ' . $key . '"><strong>' . $arrRow['headline'] . '</strong> - ' . $date . ($arrRow['addGallery'] ? ' <img src="system/modules/newsgallery/html/images.png" alt="" style="position:absolute; margin-left:10px" />' : '') . '</div>
<div class="limit_height' . (!$GLOBALS['TL_CONFIG']['doNotCollapse'] ? ' h64' : '') . ' block">
' . $arrRow['text'] . '
</div>' . "\n";
	}
}

